### ISSUE TITLE
Bug: `FlatESLint#findConfigFile` should return `undefined` if it doesn't find `eslint.config.js`

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#17150]
### Environment

Node version: v16.14.0
npm version: 8.3.1
Local ESLint version: `main` branch
Global ESLint version: 
Operating System: windows

### What parser are you using?

Default (Espree)

### What did you do?

In the `main` branch, I deleted the `eslint.config.js` file, and added the following `test.mjs` file:

```js
import unsupportedAPI from "eslint/use-at-your-own-risk";

const { FlatESLint } = unsupportedAPI;

const eslint = new FlatESLint();

console.log(await eslint.findConfigFile());


```

Then ran `node test.mjs`

### What did you expect to happen?

`FlatESLint#findConfigFile()` to return `undefined` when it doesn't find an `eslint.config.js` file.

### What actually happened?

Throws an error:

```
Error: Could not find config file.
    at locateConfigFileToUse (C:\milos\eslint\lib\eslint\flat-eslint.js:346:19)
    at async FlatESLint.findConfigFile (C:\milos\eslint\lib\eslint\flat-eslint.js:1189:36)
    at async file:///C:/milos/eslint/test.mjs:8:15
```

### Link to Minimal Reproducible Example

`main` branch

### Participation

- [ ] I am willing to submit a pull request for this issue.
